xlators/storage/posix/src/posix-inode-fd-ops.c

```diff
@@ -2519,6 +2519,47 @@ posix_flush(call_frame_t *frame, xlator_t *this, fd_t *fd, dict_t *xdata)
     return 0;
 }
 
+static int
+posix_unlink_renamed_file_and_free_ctx(xlator_t *this, inode_t *inode)
+{
+    int ret = 0;
+    char *unlink_path = NULL;
+    uint64_t ctx_uint1 = 0;
+    uint64_t ctx_uint2 = 0;
+    posix_inode_ctx_t *ctx = NULL;
+    posix_mdata_t *mdata = NULL;
+    struct posix_private *priv = this->private;
+
+    ret = inode_ctx_del2(inode, this, &ctx_uint1, &ctx_uint2);
+    if (!ctx_uint1)
+        goto check_ctx2;
```

Comment:
I think we can minimize the use of `goto` by rearranging the code a bit different. For example we can check here for `ctx_uint2` and free it if necessary and then continue with `ctx_uint1`, for example.

```c
    GF_FREE((posix_mdata_t *)(uintptr_t)ctx_uint2);

    if (ctx_uint1 == 0) {
        return 0;
    }

    ctx = (posix_inode_ctx_t *)(uintptr_t)ctx_uint1;
    /* ... */
```

Reply:
sure